### PR TITLE
partially implemented rest

### PR DESCRIPTION
**Pull request by s3ld0n**

### CODE REVIEW

File: src/main/java/com/artisan/solutions/gallery/service/defaults/ArtistServiceDefault.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.artisan.solutions.gallery.persistence.repo.ArtistRepo;
+import com.artisan.solutions.gallery.service.ArtistService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ArtistServiceDefault implements ArtistService {
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/service/defaults/ArtistServiceDefault.java:13]: Usual convention or DefaultBeanName or BeanNameImpl

File: src/main/java/com/artisan/solutions/gallery/service/Crud.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.artisan.solutions.gallery.service;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface Crud<T> {
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/service/Crud.java:6]: Crud what?

File: src/main/java/com/artisan/solutions/gallery/spring/PersistenceJpaConfig.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+public class PersistenceJpaConfig {
+
+    @Autowired
+    private Environment env;
+
+    public PersistenceJpaConfig() {
+        super();
+    }
+
+    // beans
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/spring/PersistenceJpaConfig.java:34]: remove comments

File: src/main/java/com/artisan/solutions/gallery/web/controller/ArtistController.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Artist create(@RequestBody ArtistDto artistDto) {
+        return artistService.create(DtoConverter.convertArtistDtoToEntity(artistDto));
+    }
+
+    @GetMapping("/{id}")
+    public ArtistDto findById(@PathVariable long id) {
+        return DtoConverter.convertArtistToDto(
+                artistService.findById(id).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND))
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/web/controller/ArtistController.java:34]: format lambda: each method from new line

File: src/main/java/com/artisan/solutions/gallery/web/controller/WorkController.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    @GetMapping
+    public List<WorkDto> findAll() {
+        return DtoConverter.convertAllWorksToDtos(workService.findAll());
+    }
+
+    @PutMapping
+    @ResponseStatus(HttpStatus.OK)
+    public WorkDto update(@RequestBody WorkDto workDto) {
+        return DtoConverter.convertWorkToDto(
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/web/controller/WorkController.java:45]: that would happen if workDto in request - not exist?

File: src/test/java/com/artisan/solutions/gallery/ArtistControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import static org.junit.Assert.assertThat;
+
+@SpringBootTest
```
Comment (valeriiafatova) [inline src/test/java/com/artisan/solutions/gallery/ArtistControllerTest.java:19]: Were requested Simple Unit tests for Controllers

File: src/main/java/com/artisan/solutions/gallery/persistence/model/Artist.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@Builder
+
+@Entity
+@Table(name = "artists")
+public class Artist {
```
Comment (valeriiafatova) [inline src/main/java/com/artisan/solutions/gallery/persistence/model/Artist.java:16]: artists/
works/
\/
artists/1/works
{...
...
}

File: src/main/java/com/artisan/solutions/gallery/web/controller/WorkController.java (hunk 5 above)
Comment (s3ld0n) [reply src/main/java/com/artisan/solutions/gallery/web/controller/WorkController.java:45]: new work would be created then

**Pull request merged**